core/tests/protractor_utils/StoryEditorPage.js

```diff
@@ -152,29 +151,39 @@ var StoryEditorPage = function() {
     await confirmDeleteChapterButton.click();
   };
 
-  this.createNewDestinationChapter = async function(title) {
-    await browser.actions().mouseMove(addDestinationChapterButton).perform();
-    await waitFor.elementToBeClickable(
-      addDestinationChapterButton,
-      'Add destination chapter button takes too long to be clickable.');
-    await addDestinationChapterButton.click();
-    await waitFor.visibilityOf(
-      newChapterTitleField,
-      'New Chapter modal takes too long to appear.');
-    await newChapterTitleField.sendKeys(title);
+  this.createNewChapter = async function(title) {
+    await general.scrollToTop();
+    await action.click(
+      'Create chapter button takes too long to be clickable.',
+      createChapterButton);
+    await action.sendKeys(
+      'New chapter title field', newChapterTitleField, title);
     await confirmChapterCreationButton.click();
     await general.scrollToTop();
   };
 
-  this.removeDestination = async function() {
-    await deleteDestinationChapterButton.click();
+  this.expectChaptersListToBe = async function(chapters) {
+    await this.expectNumberOfChaptersToBe(chapters.length);
+    for (var i = 0; i < chapters.length; i++) {
+      expect(await chapterTitles.get(i).getText()).toEqual(chapters[i]);
+    }
   };
 
-  this.selectDestinationChapterByName = async function(chapterName) {
-    var destinationOption = destinationSelect.element(
-      by.cssContainingText('option', chapterName));
-    await destinationOption.click();
-  };
+  this.dragChapterToAnotherChapter = (async function(chapter1, chapter2) {
+    for (var i = 0;i < await chapterTitles.count(); i++) {
+      if (await chapterTitles.get(i).getText() === chapter1) {
+        break;
+      }
+    }
+    var toMove = chapterTitles.get(i);
```

Comment:
Can we fail with a helpful error message if no match is found? That'd be clearer than an index out of bounds error

Reply:
(Added a boolean for this.)Done.

Reply:
The boolean works, though you could also just check whether `i == chapterTitles.count()`. I think either way is fine though